### DISCUSSION TITLE
How to Increase the Figsize of Plot or Diagram?

### DISCUSSION THREAD

**Discussion by Jerry2001** [colour-science/colour#906] [Q&A] [answered]
Hello everyone, I'm trying to increase the figure size of the chromaticity diagram but haven't found much luck. I [tried](https://ibb.co/Jt4K3Qn) to insert `figure=plt.figure(figsize=(20, 20))` in `plot_chromaticity_diagram_CIE1931`. But, the resulting figure size has 0 axes and the saved file is just a transparent file. Thank you so much!

**Comment by thomasmansencal** [comment, marked as answer]
Hi @Jerry2001,

Great questions, there are a few ways to do it! All the *Colour* plotting definitions starts by creating a figure and an axes with the [colour.plotting.artist](https://github.com/colour-science/colour/blob/6829b363d5f0682bff0f4826995e7ceac189ff28/colour/plotting/common.py#L421) definition which can either use an existing axes or create a new figure using `plt.rcParams['figure.figsize']`.

With that in mind:

### Defaults

```python
colour.plotting.plot_chromaticity_diagram_CIE1931()
```

![image](https://user-images.githubusercontent.com/99779/142691774-39377b69-10f4-41d2-97ce-8831a0bd461a.png)

### Using an Existing Axes

```python
figure = plt.figure(figsize=(19.2, 10.8))

colour.plotting.plot_chromaticity_diagram_CIE1931(axes=figure.gca())
```

![image](https://user-images.githubusercontent.com/99779/142691841-80629306-d9c8-437d-840d-f1da76f65315.png)

### Using a Temporary Override

```python
with plt.style.context({'figure.figsize': (12.80, 7.20)}):
    colour.plotting.plot_chromaticity_diagram_CIE1931()
```

![image](https://user-images.githubusercontent.com/99779/142691978-52a9fc04-41c7-4c60-9e3d-a89a630d2f89.png)

### Using a Definition with the Colour Decorator

```python
@colour.plotting.override_style(**{'figure.figsize': (8, 6)})
def plot_custom_chromaticity_diagram(**kwargs):
    return colour.plotting.plot_chromaticity_diagram_CIE1931(**kwargs)

plot_custom_chromaticity_diagram()
```

![image](https://user-images.githubusercontent.com/99779/142692319-63002387-513c-4775-bf1a-4af903c4fb9f.png)

This [Google Colab Notebook](https://colab.research.google.com/drive/1HebFPqvKtFbdFIKa4gzFUc2VLDGhGXNq?usp=sharing) has the above code and defaults plots to show that the state is restored properly after each custom figure size.

**Reply by Jerry2001** [reply]
Hi @KelSolaar, thank you for your prompt and thorough answer. I appreciate it a lot!
